### PR TITLE
[pkg/telemetryquerylanguage] Move contexts to package

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
This PR copies the contexts from the transform processor into packages in the telemetryquerylanguage module.

**Link to tracking Issue:** 
#11751

**Testing:** 
Ran unit tests

**Documentation:** 
Added new READMEs

### CODE REVIEW

File: pkg/telemetryquerylanguage/contexts/metrics/metrics_test.go
Comment (kentquirk) [inline pkg/telemetryquerylanguage/contexts/metrics/metrics_test.go:38]: I know there's nothing technically wrong with using `new` as a field name (here and in other test files), but it's a bit jarring as a field name, and it confuses GitHub's syntax colorer. Would you consider renaming this field to, say, `newval` or even `newv`?

File: pkg/telemetryquerylanguage/contexts/metrics/README.md
Comment (kentquirk) [inline pkg/telemetryquerylanguage/contexts/metrics/README.md:36]: Maybe along with using the proto defs, we could link these docs to the appropriate proto defs file and not duplicate the values here? We'd like to avoid getting bitten by this stuff.

**Review by kentquirk — COMMENTED** [review]
A few nits, but this seems quite close. Sorry for the delay.

File: pkg/telemetryquerylanguage/contexts/logs/logs.go
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/logs/logs.go:62]: I like it, I'll try it

File: pkg/telemetryquerylanguage/contexts/metrics/metrics_test.go
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/metrics_test.go:38]: No problem

File: pkg/telemetryquerylanguage/contexts/metrics/README.md
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/README.md:36]: I'll link to the protos in this section as well.

File: pkg/telemetryquerylanguage/contexts/logs/logs.go
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/logs/logs.go:62]: Was able to get all but FLAG_NONE and FLAG_NO_RECORDED_VALUE.  Annoyingly those aren't includes in the metrics proto library and aren't exposed in pdata.

File: pkg/telemetryquerylanguage/contexts/metrics/README.md
Comment (TylerHelmuth) [reply pkg/telemetryquerylanguage/contexts/metrics/README.md]: I linked to the protos in each README's enum section.  For metrics specifically I linked to the pdata numeric definitions for the data type enum, but we need to keep the table because the actual Enum Symbol is defined by the Metrics Context.

**Review by kentquirk — APPROVED** [review]
Thank you!

**Comment by TylerHelmuth** [comment]
@open-telemetry/collector-contrib-approvers please review

**Comment by TylerHelmuth** [comment]
@dashpole are you free to review?  There are a lot of changes, but they are copy and paste from the transform processor.  See @kentquirk's comments for anything that was modified.

**Comment by TylerHelmuth** [comment]
With the merging of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12597 the contexts being added in this PR will need updated with the StandardGetSetter, but to avoid logic changes in this PR I will submit a different PR to do that update once this is merged.

**Review by dashpole — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**
